nydus-image：optimize chunkdict save

cslinwang opened the pull request:
## Relevant Issue (if applicable)
_If there are Issues related to this PullRequest, please list it._

## Details
Refactor the Deduplicate implementation to only
initialize config when inserting chunk data.
Simplify code for better maintainability.


## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

imeoer commented on src/bin/nydus-image/deduplicate.rs:
```diff
@@ -195,7 +204,11 @@ impl Deduplicate<SqliteDatabase> {
             Ok(())
         };
-        let tree = Tree::from_bootstrap(&self.sb, &mut ())
+        let sb = self
```
How about `fn insert_chunks(&mut self, sb: &RafsSuper) -> anyhow::Result<()> {`

cslinwang replied:
OK

cslinwang committed:
nydus-image：Optimize Chunkdict Save

Refactor the Deduplicate implementation to only
initialize config when inserting chunk data.
Simplify code for better maintainability.

Signed-off-by: Lin Wang <l.wang@mail.dlut.edu.cn>

An-DJ commented on src/bin/nydus-image/main.rs:
```diff
@@ -399,7 +398,7 @@ fn prepare_cmd_args(bti_string: &'static str) -> App {
                     .arg(arg_output_json.clone())
             )
-    );
+                );
```
Can't believe that cargo fmt didn't find this...

imeoer replied:
It seems `cargo fmt` is broken from L164. 😂
